wasm-text.md

```diff
@@ -360,6 +360,233 @@ Imports can be declared like regular functions, memories, etc., by giving an inl
     rule ( memory OID:OptionalId (import MOD NAME) MT:MemType   ) => ( import MOD NAME (memory OID MT  ) )  [macro]
 ```
 
+Desugaring
+----------
+
+The text format is one of the concrete formats of Wasm.
+Every concrete format maps to a common structure, described as an abstract syntax.
+The function `text2abstract` is a partial function which maps valid programs in the text format to the abstract format.
+Some classes of invalid programs, such as those where an identifier appears in a context in which it is not declared, are undefined.
+
+The function deals with the desugarings which are context dependent.
+Other desugarings are either left for runtime or expressed as macros (for now).
+
+TODO:
+
+* Get rid of inline type declarations (only allow types defined first, inline type declarations serve as documentation and identifier bindings). Something like `(func (type X) TDS:TDecls ... ) => (func (type X))` and `(func TDS:TDecls ...) => (type TDECLS) (func (type NEXT_TYPE_ID) or something)`
+* Remove module names.
+* Give the text format and core format different types, and have the preprocessing handle the conversion. So that identifiers don't even exist in the core type.
```

Comment:
The first bullet point here is a bit confusing, can you be more clear?

Reply:
Sure, how about this:

```suggestion
* Get rid of inline type declarations. The text format allows specifying the type directly in the function header using the `param` and `result` keywords. However, these will either be desugared to a new top-level `type` declaration or they must match an existing one. In the abstract format, a function's type is a pointer to a top-level `type` declaration. This could either be an immediate desugaring, like `(func (type X) TDS:TDecls ... ) => (func (type X))` and `(func TDS:TDecls ...) => (type TDECLS) (func (type NEXT_TYPE_ID)`, or an initial pass could gather all the function types and declare them)
* Remove module names.
* Give the text format and core format different types, and have the preprocessing handle the conversion. So that identifiers don't even exist in the core type.
```

Reply:
Looks better for sure. My only nit is that I prefer `-   ` instead of `* ` (so dash followed by three spaces) for bullet lists. And I prefer one sentence per line.